packages/edit-site/src/components/post-edit/index.js

```diff
@@ -123,14 +131,39 @@ function PostEditForm( { postType, postId } ) {
 		setMultiEdits( {} );
 	}, [ ids ] );
 
+	const { ExperimentalBlockEditorProvider } = unlock(
+		blockEditorPrivateApis
+	);
+	const settings = usePatternSettings();
+
+	/**
+	 * The template field depends on the block editor settings.
+	 * This is a workaround to ensure that the block editor settings are available.
+	 */
+	const fieldsWithDependency = useMemo( () => {
+		return fields.map( ( field ) => {
+			if ( field.id === 'template' ) {
+				return {
+					...field,
+					Edit: ( data ) => (
+						<ExperimentalBlockEditorProvider settings={ settings }>
```

Comment:
Why is this here and not directly in the template field edit?

Reply:
Not asking to change it, just curious why

Reply:
I need to pass `settings` to this provider and `usePatternSettings` depends on the edit site store:

https://github.com/WordPress/gutenberg/blob/f891475bd3f69bcb66b1fdadff1e7173fdf5d424/packages/edit-site/src/components/page-patterns/use-pattern-settings.js#L15-L20

Reply:
Thanks! I agree with you! I'm happy to:
- open an issue.
- add the issue link on this comment.
- dedicate some time in the upcoming weeks to fix this.